### PR TITLE
match templated paths and return matched variables

### PR DESCRIPTION
**Pull request by theganyo**
match templated paths and return matched variables

### CODE REVIEW

**Comment by rockspore** [comment]
I tested this test case:
```
{path: "a/x/x/x/c/x/c/c", value: "a/{b=**}/x/c/{d=**}/c", values: map[string]interface{}{"b": "x/x", "d": "x/c"}},
```
and it resulted in `"b": "x/x/x"`.

**Comment by theganyo** [comment]
That's not a valid case unless you added `a/{b=**}/x/c/{d=**}/c` to the tree creation as well. But if you do, you'll break the last test - as yours seems to be basically a duplication of that one.

File: path/tree.go [hunk 1]
```diff
@@ -93,49 +113,64 @@ func (t *tree) Find(path []string, index int) interface{} {
...
-	node, _ := t.findNode(path, index, 0)
+	node, _ := t.findNode(path, index, 0, nil)
 	if node != nil {
 		return node.value
 	}
 	return nil
 }
 
-func (t *tree) findNode(path []string, index, matchCount int) (found *tree, foundMatchCount int) {
+func (t *tree) FindAndExtract(path []string, index int) (val interface{}, varMap map[string]interface{}) {
```
Comment (rockspore) [inline path/tree.go:123]: Comment for the method.

Comment (theganyo) [reply path/tree.go:123]: It's right there on the interface, but ok, I'll repeat it. :)

**Review by rockspore — APPROVED** [review]

**Pull request merged**